server/src-lib/Hasura/GraphQL/Execute/Remote.hs

```diff
@@ -0,0 +1,46 @@
+module Hasura.GraphQL.Execute.Remote
+  ( buildTypedOperation
+  ) where
+
+import           Hasura.Prelude
+
+import qualified Data.HashMap.Strict                    as Map
+import qualified Language.GraphQL.Draft.Syntax          as G
+import qualified Data.HashSet                           as Set
+
+import qualified Hasura.GraphQL.Transport.HTTP.Protocol as GH
+
+import           Hasura.GraphQL.Execute.Prepare
+import           Hasura.GraphQL.Parser
+
+unresolveVariables
+  :: forall fragments
+   . Functor fragments
+  => G.SelectionSet fragments Variable
+  -> G.SelectionSet fragments G.Name
+unresolveVariables =
+  fmap (fmap (getName . vInfo))
+
+collectVariables
+  :: forall fragments var
+   . (Foldable fragments, Hashable var, Eq var)
+  => G.SelectionSet fragments var
+  -> Set.HashSet var
+collectVariables =
+  Set.unions . fmap (foldMap Set.singleton)
+
+buildTypedOperation
```

Comment:
This is subjective, but I found the name `buildTypedOperation` a little bit mystifying, though I suppose in some sense it is ultimately what the function does. I think I would find a name like `buildExecStepRemote` more helpful.

Reply:
You're absolutely right, this method should've been named after its role, not after its implementation.